### DISCUSSION TITLE
How does Marker works when using XML and not compose

### DISCUSSION THREAD

**Discussion by DannyBloky** [patrykandpatrick/vico#524] [Questions] [answered]
I had a question about the markers.

I don't understand the documentation about it how I can use the markers.

I'm using version 1.13.1

I'm using this chart:
` <com.patrykandpatrick.vico.views.chart.ChartView
            android:id="@+id/lineChart"
            android:layout_width="match_parent"
            android:layout_height="200dp"
            app:chart="line"
            app:showBottomAxis="true"
            app:showStartAxis="true" />`

This is inside my Fragment now
`override fun onViewCreated(view: View, savedInstanceState: Bundle?) {
        super.onViewCreated(view, savedInstanceState)
        observeBaseViewModel(viewModel)

        val binding = FragmentBinding.bind(view)
        val endColor = getColorWithPercentAlpha(requireContext(), R.color.backgroundColorBlue, 0.05)
        val startColor =
            getColorWithPercentAlpha(requireContext(), R.color.backgroundColorBlue, 0.7)

        viewModel.chartData.observe(viewLifecycleOwner) { value ->
            val entryModelProducer = ChartEntryModelProducer()

            val list = mutableListOf<FloatEntry>()
            for (item in value) {
                list.add(
                    FloatEntry(
                        item.receivedAtDate.toEpochSecond(ZoneOffset.UTC).toFloat(),
                        item.value.toFloat()
                    )
                )
            }
            entryModelProducer.setEntries(list)
            binding.lineChart.entryProducer = entryModelProducer
        }

        with(binding) {
            //Chart
            with(binding.lineChart.bottomAxis as HorizontalAxis) {
                valueFormatter = bottomAxisValueFormatter
                itemPlacer = startAxisXItemPlacer
                label = textComponent {
                    color = R.color.gray_date
                    textSizeSp = 12f
                }
                axisLine?.color = Color.BLACK
                guideline?.color = Color.TRANSPARENT
            }

            with(binding.lineChart.chart as LineChart) {
                addDecorations(
                    listOf(
                        ThresholdLine(
                            20000f,
                            "",
                            lineComponent = ShapeComponent(
                                color = Color.RED
                            )
                        ),
                        ThresholdLine(
                            25000f,
                            "",
                            lineComponent = ShapeComponent(
                                color = Color.RED
                            )
                        )
                    )
                )

                lines[0].lineColor = R.color.backgroundColorBlue
                lines[0].lineBackgroundShader =
                    DynamicShaders.verticalGradient(startColor, endColor)
            }

            with(binding.lineChart.startAxis as VerticalAxis) {
                itemPlacer = startAxisYItemPlacer
                label = textComponent {
                    color = R.color.gray_date
                    textSizeSp = 12f
                }
                axisLine?.color = Color.BLACK
                guideline?.color = Color.TRANSPARENT
            }
        }
    }`
            
         
  This code will show a line chart with 2 tresholds and a line graph with gradient area.
  Also custom labels on my x line with dates. 
  
  But I want also have the click function on the chart that show the marker. But I don't understand or is this not possible in xml. I don't see a property like enable marker or something.
  
  Only thing is Marker and add addPersistentMarker
  But as far I can see is that just 1 marker on a specific place. Also don't get that to work either. So if someone can help me with this

**Comment by patrickmichalik** [comment]
Hello! This is described in the wiki. Please see the second paragraph [here](https://patrykandpatrick.com/vico/wiki/extra-elements#marker).

**Reply by DannyBloky** [reply]
Thanks for your reply. I have read the documentation but I didn't understand it. So if you can help me that would be great how to do it.

**Reply by csbenz** [reply]
@DannyBloky Is LineChart part of this library? I can't find it, I'm trying to set the color of my lines.

**Reply by patrickmichalik** [reply, marked as answer]
You can add a standard `Marker` by updating `lineChart.marker`, @DannyBloky. For example, if you’re creating your `Marker` via `MarkerComponent`, use the following:

```kt
lineChart.marker = MarkerComponent(/* ... */)
```

By the way, it looks like you’re recreating your `ChartEntryModelProducer` on every data update. This is discouraged and may cause unexpected behavior. I’d recommend using a single `ChartEntryModelProducer` and storing it at a higher level.

@csbenz, `LineChart` is a Vico 1 class. In Vico 2, use `LineCartesianLayer`.

**Reply by DannyBloky** [reply]
Thank you I now got the text of the Marker. If I got any question later about it I will ask them. 

Good point. I saw it when I was changing the data. Then the chart will be blank. So I did your suggestion for the `ChartEntryModelProducer ` and now it can update the data.

**Reply by patrickmichalik** [reply]
I’m glad to hear that, @DannyBloky. You’re welcome!

**Comment by DannyBloky** [comment]
> @DannyBloky Is LineChart part of this library? I can't find it, I'm trying to set the color of my lines.


lines[0].lineColor = R.color or Color.
It in the with(binding.lineChart.chart as LineChart) 

By this you can set the color of the line
